test/wallet-test.js

```diff
@@ -1177,6 +1177,49 @@ describe('Wallet', function() {
     assert.strictEqual(t2.getFee(), 10000);
   });
 
+  it('should use the change address in the options', async () => {
+    const alice = await wdb.create();
+
+    const receive = Address.fromPubkeyhash(Buffer.alloc(20, 1));
+    const change = Address.fromPubkeyhash(Buffer.alloc(20, 2));
+
+    // Coinbase
+    const t1 = new MTX();
+    t1.addInput(dummyInput());
+    t1.addOutput(await alice.receiveAddress(), 5460);
+    t1.addOutput(await alice.receiveAddress(), 5460);
+    t1.addOutput(await alice.receiveAddress(), 5460);
+    t1.addOutput(await alice.receiveAddress(), 5460);
+
+    await wdb.addTX(t1.toTX());
+
+    const options = {
+      subtractFee: true,
+      rate: 0,
+      sort: false,
+      round: true,
+      outputs: [{address: receive, value: 21000}],
+      changeAddress: change
+    };
+
+    // Create new transaction
+    const t2 = await alice.createTX(options);
+    await alice.sign(t2);
+
+    assert(t2.verify());
+
+    const total = t2.inputs
+      .map(i => t2.view.getCoinFor(i))
+      .reduce((acc, coin) => acc + coin.value, 0);
```

Comment:
Since `t2` is actually a `MTX` object here, this algorithm has already been written for you :-)

`const total = t2.getInputValue();`

See:

https://github.com/bcoin-org/bcoin/blob/f991b0a60c575610cdbcdfe2205316a2012a61f5/lib/primitives/mtx.js#L304-L306

and then:

https://github.com/bcoin-org/bcoin/blob/f991b0a60c575610cdbcdfe2205316a2012a61f5/lib/primitives/tx.js#L998-L1011

Reply:
I agree, that's much better,  will sort it out.  But just FYI, you do realise I did that after you suggested the following in your previous comment right ? :)

 
```js
let total = 0;
for (const input of tx.inputs)
   total += input.coin.value;
```

Reply:
Yup. Sorry! I was triggered by the `.map` and the `.reduce`... not our usual style, made me look in to it more :-)